gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -664,14 +664,16 @@ public GsonBuilder setDateFormat(int dateStyle, int timeStyle) {
    * @param typeAdapter This object must implement at least one of the {@link TypeAdapter},
    * {@link InstanceCreator}, {@link JsonSerializer}, and a {@link JsonDeserializer} interfaces.
    * @return a reference to this {@code GsonBuilder} object to fulfill the "Builder" pattern
+   * @throws IllegalArgumentException if the Type adapter being registered is for {@code Object} class or {@link JsonElement} or any of its subclasses
```

Comment:
Is there a reason why you capitalize "Type" (instead of "type")? It is a bit inconsistent with the previous sentences.